src/XML/src/FMI3/fmi3_xml_type.c

```diff
@@ -68,245 +68,250 @@ fmi3_xml_int_typedef_t* fmi3_xml_get_type_as_int(fmi3_xml_variable_typedef_t* t)
-    fmi3_xml_variable_typedef_t* vt = (void*)t;
-    fmi3_xml_float_type_props_t* props = (fmi3_xml_float_type_props_t*)(vt->super.nextLayer);
-    return props->typeMin.scalar32s;
+    return fmi3_xml_get_float_type_props(t)->typeMin.scalar32s;
 }
 
 fmi3_float32_t fmi3_xml_get_float32_type_max(fmi3_xml_float_typedef_t* t) {
-    fmi3_xml_variable_typedef_t* vt = (void*)t;
-    fmi3_xml_float_type_props_t* props = (fmi3_xml_float_type_props_t*)(vt->super.nextLayer);
-    return props->typeMax.scalar32s;
+    return fmi3_xml_get_float_type_props(t)->typeMax.scalar32s;
 }
 
 fmi3_float32_t fmi3_xml_get_float32_type_nominal(fmi3_xml_float_typedef_t* t) {
-    fmi3_xml_variable_typedef_t* vt = (void*)t;
-    fmi3_xml_float_type_props_t* props = (fmi3_xml_float_type_props_t*)(vt->super.nextLayer);
-    return props->typeNominal.scalar32s;
+    return fmi3_xml_get_float_type_props(t)->typeNominal.scalar32s;
 }
 
 fmi3_xml_unit_t* fmi3_xml_get_float32_type_unit(fmi3_xml_float_typedef_t* t) {
-    fmi3_xml_variable_typedef_t* vt = (void*)t;
-    fmi3_xml_float_type_props_t* props = (fmi3_xml_float_type_props_t*)(vt->super.nextLayer);
-    fmi3_xml_display_unit_t* du = props->displayUnit;
-    if(du) return du->baseUnit;
-    return 0;
+    fmi3_xml_display_unit_t* du = fmi3_xml_get_float_type_props(t)->displayUnit;
+    if (du) {
+        return du->baseUnit;
+    }
+    return NULL;
 }
 
 int fmi3_xml_get_float32_type_is_relative_quantity(fmi3_xml_float_typedef_t* t) {
-    fmi3_xml_variable_typedef_t* vt = (void*)t;
-    fmi3_xml_float_type_props_t* props = (fmi3_xml_float_type_props_t*)(vt->super.nextLayer);
-    return props->super.isRelativeQuantity;
+    return fmi3_xml_get_float_type_props(t)->super.isRelativeQuantity;
 }
 
 int fmi3_xml_get_float32_type_is_unbounded(fmi3_xml_float_typedef_t* t) {
-    fmi3_xml_variable_typedef_t* vt = (void*)t;
-    fmi3_xml_float_type_props_t* props = (fmi3_xml_float_type_props_t*)(vt->super.nextLayer);
-    return props->super.isUnbounded;
+    return fmi3_xml_get_float_type_props(t)->super.isUnbounded;
 }
 
 fmi3_xml_display_unit_t* fmi3_xml_get_float32_type_display_unit(fmi3_xml_float_typedef_t* t) {
-    fmi3_xml_variable_typedef_t* vt = (void*)t;
-    fmi3_xml_float_type_props_t* props = (fmi3_xml_float_type_props_t*)(vt->super.nextLayer);
-    fmi3_xml_display_unit_t* du = props->displayUnit;
-    if(!du || (&du->baseUnit->defaultDisplay == du)) return 0;
+    fmi3_xml_display_unit_t* du = fmi3_xml_get_float_type_props(t)->displayUnit;
+    if (!du || (&du->baseUnit->defaultDisplay == du)) {
+        return NULL;
+    }
     return du;
 }
 
+// -----------------------------------------------------------------------------
+// Int64
+// -----------------------------------------------------------------------------
 
-fmi3_int64_t fmi3_xml_get_int64_type_min(fmi3_xml_int_typedef_t* t) {
+static fmi3_xml_int_type_props_t* fmi3_xml_get_int_type_props(fmi3_xml_int_typedef_t* t) {
```

Comment:
Otherwise I'll need to move it way up above FloatXX, which would break the pattern of having it under the "header" like for the others. I could also make an IntXX one, but the headers are mainly to allow some quick visual separation of the code, given its repetitive nature.